webrtc/RTCPeerConnection-createDataChannel.html

```diff
@@ -29,11 +29,70 @@
   assert_equals(channel.maxRetransmits, null, 'maxRetransmits');
   assert_equals(channel.protocol, '', 'protocol');
   assert_equals(channel.negotiated, false, 'negotiated');
-  // Initial id value is not defined.
-  assert_equals(typeof channel.id, 'number', 'id type');
+  // Since no offer/answer exchange has occurred yet, the DTLS role is unknown
+  // and so the ID should be null.
+  assert_equals(channel.id, null, 'id');
   assert_equals(channel.priority, 'low', 'priority');
 }, 'createDataChannel defaults');
 
+async_test(test => {
+  const pc = new RTCPeerConnection;
+  const channel = pc.createDataChannel('');
+  pc.createOffer()
+  .then(function(offer) {
+    return pc.setLocalDescription(offer);
+  })
+  .then(function() {
+    // Turn our own offer SDP into valid answer SDP by setting the DTLS role to
+    // "active".
+    const answer = new RTCSessionDescription({
+      type: "answer",
+      sdp: pc.localDescription.sdp.replace("actpass", "active")
+    });
+    return pc.setRemoteDescription(answer);
+  })
+  .then(function() {
+    // Since the remote description had an "active" DTLS role, we're the server
+    // and should use odd data channel IDs, according to rtcweb-data-channel.
+    assert_equals(channel.id, 1, 'id');
```

Comment:
I don't think the spec says this must be 1, just that it must be odd:
"If the side is the DTLS client, it MUST choose an even
   Stream Identifier, if the side is the DTLS server, it MUST choose an
   odd one.  "